### ISSUE TITLE
libext2fs: compile errors with current Genode staging branch

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode-world#167]
When building 'make LIB=libext2fs' with the current Genode staging branch, the following error occurs:

```
    COMPILE  alloc.o
In file included from /.../contrib/fuse-ext2-867fc9cb8f828f6463871d5db396b8cda9d7b5b5/src/lib/fuse-ext2/e2fsprogs-1.41.12.newgit/ext2fs/alloc.c:26:
/.../contrib/fuse-ext2-867fc9cb8f828f6463871d5db396b8cda9d7b5b5/src/lib/fuse-ext2/e2fsprogs-1.41.12.newgit/ext2fs/ext2fs.h:52:22: error: conflicting types for ‘dev_t’
 typedef unsigned int dev_t;
                      ^~~~~
In file included from /.../contrib/fuse-ext2-867fc9cb8f828f6463871d5db396b8cda9d7b5b5/src/lib/fuse-ext2/e2fsprogs-1.41.12.newgit/ext2fs/ext2fs.h:49,
                 from /.../contrib/fuse-ext2-867fc9cb8f828f6463871d5db396b8cda9d7b5b5/src/lib/fuse-ext2/e2fsprogs-1.41.12.newgit/ext2fs/alloc.c:26:
/.../contrib/libc-15bedbc9f3adea283d1c036ef57448ebbdaa4cbc/include/libc/sys/types.h:107:17: note: previous declaration of ‘dev_t’ was here
 typedef __dev_t  dev_t;  /* device number or struct cdev */
```

**Comment by ttcoder** [comment]
I don't have the code handy, but I think I "solved" that one by removing the typedef, and replacing it with ...

`#include <sys/types>`

...So that the proper definition is used instead.

But since this is in contrib/, the thing to do is probably to create a .patch file (to be applied by tools/prepare_port) rather than change the code by hand ?

**Comment by cnuke** [comment]
Problem was fixed (see #199).
